src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -57,11 +76,12 @@ private YouTube getService() throws GeneralSecurityException, IOException {
       System.err.println(e.getDetails().getMessage());
       
       for (GoogleJsonError.ErrorInfo err : e.getDetails().getErrors()) {
-        if (!err.getReason().equals("commentsDisabled")) { // Ex. invalid video id or private video
-          throw new IllegalArgumentException();
-        } else { // Video with disabled comments
+        if (err.getReason().equals("videoNotFound")) { // Invalid video id or private video
+          throw new IllegalArgumentException("Video is private or does not exist.");
+        } else if (err.getReason().equals("commentsDisabled")) { // Video with disabled comments
           return new ArrayList<>();
         }
+        throw e;
```

Comment:
Move this after the for loop. Maybe other errors in the error list can have video not found or comments disabled.